disttask/framework/dispatcher/dispatcher.go

```diff
@@ -206,12 +223,75 @@ func (d *dispatcher) onRunning() error {
 		logutil.Logger(d.logCtx).Info("previous stage finished, generate dist plan", zap.Int64("stage", d.task.Step))
 		return d.onNextStage()
 	}
+	// Check if any node are down.
+	handle := GetTaskFlowHandle(d.task.Type)
+	if err := d.replaceDeadNodesIfAny(handle); err != nil {
+		return err
+	}
 	// Wait all subtasks in this stage finished.
-	GetTaskFlowHandle(d.task.Type).OnTicker(d.ctx, d.task)
+	handle.OnTicker(d.ctx, d.task)
 	logutil.Logger(d.logCtx).Debug("on running state, this task keeps current state", zap.String("state", d.task.State))
 	return nil
 }
 
+func (d *dispatcher) replaceDeadNodesIfAny(handle TaskFlowHandle) error {
+	if len(d.taskNodes) == 0 {
+		schedulerIDs, err := d.taskMgr.GetSchedulerIDsByTaskID(d.task.ID)
+		if err != nil {
+			return err
+		}
+		d.taskNodes = schedulerIDs
+	}
+	d.liveNodeFetchTick++
+	if d.liveNodeFetchTick == d.liveNodeFetchInterval {
+		d.liveNodeFetchTick = 0
+		serverInfos, err := GenerateSchedulerNodes(d.ctx)
+		if err != nil {
+			return err
+		}
+		eligibleServerInfos, err := handle.GetEligibleInstances(d.ctx, d.task)
+		if err != nil {
+			return err
+		}
+		newInfos := serverInfos[:0]
+		for _, m := range serverInfos {
+			found := false
+			for _, n := range eligibleServerInfos {
+				if m.ID == n.ID {
+					found = true
+					break
+				}
+			}
+			if found {
+				newInfos = append(newInfos, m)
+			}
+		}
+		d.liveNodes = newInfos
+	}
+	if len(d.liveNodes) > 0 {
+		replaceNodes := make(map[string]string)
+		for _, nodeID := range d.taskNodes {
+			if ok := disttaskutil.MatchServerInfo(d.liveNodes, nodeID); !ok {
+				n := d.liveNodes[rand.Int()%len(d.liveNodes)] //nolint:gosec
```

Comment:
Maybe add `rand.Rand` as a member of `*dispatcher` to prevent the influence of `rand.Seed`.

```
randSrc := rand.NewSource(time.Now().UnixNano())
dispatcher.rand = rand.New(randSrc)
```